### DISCUSSION TITLE
Allow skipping the junction table in m2m retrieve

### DISCUSSION THREAD

**Discussion by davidbwaters** [directus/directus#4112] [Feature Requests]
Hi, when using 'items' with the API for multiple files, it returns the collection ID.  If it's a single file, it returns the file ID.  The junction isn't configurable for multiple files.  I would think the expected behavior and best for most use cases would be to return the file ID like when referencing media directly.  Otherwise I have to use fields and a bunch of extra query params or make another query to the other collection. 

I searched and couldn't find another issue for this. 

If I use a relational collection to another collection of files, the file ID is returned as expected.  Unless I'm missing something, is there a straightforward way to return the image ID for multiple images or can this be an option?

On latest RC 38 on Digital Ocean, Caddy server.

**Comment by rijkvanzanten** [comment]
This is the same as any other many-to-many relationship in Directus. There's currently no way to "skip" the junction table in the API

**Reply by davidbwaters** [reply]
I really just want to set the primary junction key ID to the file ID so that's what's returned when I read items.

**Reply by davidbwaters** [reply]
Originally I think I was thrown off because I expected this to behave like a one-to-many and because a single image returns the file id.  I'm curious as to why this is a many-to-many and not a one-to-many.

That's one reason I was expecting the file id.  A one-to-many of files returns the file ID by default.  I would think for most use cases, a file UID would be much more useful, since it's unique and required to be able to load the asset.

**Reply by rijkvanzanten** [reply]
> I'm curious as to why this is a many-to-many and not a one-to-many.

Because you'd generally want files to be able to be associated to multiple other collection types, without ending up with countless additional foreign key constraints on the file itself.

> I would think for most use cases, a file UID would be much more useful, since it's unique and required to be able to load the asset.

Yeah, I'm not arguing against that. It's not uncommon to have additional fields on the junction table, so we can't just strip out the junction table at all times, hence me considering your discussion here a feature request to include have a way to skip the junction table which in turn would immediately return the related files 👍🏻

**Reply by davidbwaters** [reply]
Gotcha.  I think also having an option to set the junction primary key to the file key would make sense.  That way it would be more consistent with the way image references would be more consistent and a file id would be returned with bare .items()  queries would be enough to get what you need from a collection.  This would be a huge boost for DX, I think.

**Reply by rijkvanzanten** [reply]
That's an interesting approach! Seeing that the primary key has to be unique, it'd mean that you could only use an image once per collection, which isn't ideal for most 🤔

**Reply by davidbwaters** [reply]
🤔 Yeah, problems like this remind me why I'm a frontend dev.  I wonder if @benhaynes has any ideas or workarounds for how to achieve this.  There was another post that was discussing the ability to set the default value the API returns to the primary display field, but if that were a separate option it would make it much easier to build frontends.

I love the thoughtful configuration options of the new Directus and think a few more tweaks could make it much more usable. From a front-end perspective, right now to get just file ID from image fields, relational, and multi-file fields from an endpoint I have to do a fetch nested fields 3 layers deep, loop through and check for filename_disk, conditionally check for directus_file_id, and use ID if thats false.  That's a lot of overhead and work just to get a file ID mapped to the field name in JSON, which is what's needed 90% of the time.  And in all this data I'm looping through, the field type would at least help figure out which data to look for.  Just figuring out whether it's a file field isn't straightforward.

I'm convincing myself GraphQL might not be overkill for me, but I still think this could be improved.

**Reply by benhaynes** [reply]
I think Rijk's idea for `skip` would help solve this. Additionally, I had an idea of "fields returned by default" that might make it easier to define what info is returned on `/items` without having to always define it using the `fields` param.

https://github.com/directus/directus/discussions/4128

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
